onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -81,77 +81,83 @@ Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) co
     }
   }
 
-  std::vector<int64_t> element_counts(last_indice_dimension, 0LL); // Number of elements for each input dimension
+  std::vector<int64_t> element_counts(last_indice_dimension, 0LL);  // Number of elements for each input dimension
 
-#ifdef _OPENMP
-#pragma omp parallel for
-#endif
   for (int64_t i = 0; i < last_indice_dimension; ++i) {
     element_counts[i] = input_shape.SizeFromDimension(i + 1);
   }
 
   int64_t err_indice = 0;
-  p.element_bytes    = input_tensor->DataType()->Size();
-  p.element_to_copy  = input_shape.SizeFromDimension(last_indice_dimension);
-  p.bytes_to_copy    = p.element_bytes * p.element_to_copy;
+  p.element_bytes = input_tensor->DataType()->Size();
+  p.element_to_copy = input_shape.SizeFromDimension(last_indice_dimension);
+  p.bytes_to_copy = p.element_bytes * p.element_to_copy;
   auto indice_offset = static_cast<const Tind*>(indice_tensor->DataRaw());
-  auto offset_count  = indice_shape.Size() / last_indice_dimension; // Times to copy
+  auto offset_count = indice_shape.Size() / last_indice_dimension;  // Times to copy
   p.element_offsets.assign(offset_count, 0LL);
 
   if (input_tensor->IsDataTypeString()) {
-    p.input_str_base  = static_cast<const std::string*>(update_tensor->DataRaw());
+    p.input_str_base = static_cast<const std::string*>(update_tensor->DataRaw());
     p.output_str_base = static_cast<std::string*>(output_tensor->MutableDataRaw());
   } else {
-    p.input_base      = static_cast<const uint8_t*>(update_tensor->DataRaw());
-    p.output_base     = static_cast<uint8_t*>(output_tensor->MutableDataRaw());
+    p.input_base = static_cast<const uint8_t*>(update_tensor->DataRaw());
+    p.output_base = static_cast<uint8_t*>(output_tensor->MutableDataRaw());
   }
 
-#ifdef _OPENMP
-#pragma omp parallel for
-#endif
-  for (int64_t i = 0; i < offset_count; ++i) {
+  auto lambda = [&](int64_t i) {
     for (int64_t j = 0; j < last_indice_dimension; ++j) {
       auto indice = *(indice_offset + i * last_indice_dimension + j);
       if (indice < 0 || indice >= input_shape[j]) {
         err_indice = indice;
       }
       p.element_offsets[i] += indice * element_counts[j];
     }
```

Comment:
Does this need to be parallelized? I suspect there's not enough work and doing so will be slower.

Reply:
Didn't seem to make much of a difference. Removed it.